### PR TITLE
feat: add node-only release

### PR DESCRIPTION
**Pull request by cristiand391**
<strike>
DEPENDS ON https://github.com/jsforce/jsforce/pull/1369
</strike>

#1369 was merged into `3.0` ✅ 

See: https://github.com/jsforce/jsforce/issues/1374

@W-14322939@

### CODE REVIEW

File: DEVELOPING.md [hunk 1]
```diff
@@ -1,21 +1,49 @@
...
+npm run jsforce-node:dev
+npm run build:node:cjs
+```
+
+**IMPORTANT:**
+
+`npm run jsforce-node:dev` will only change the `name` field in the `package.json` to `@jsforce/jsforce-node`.
+This helps reducing changes to the package.json (smaller diff) and allows you to `yarn | npm` link it, but you should never commit the change back to the repo.
+
+You could also run `npm run jsforce-node` instead to get the full changes we do at release time, see: [JSFORCE_NODE.md](./JSFORCE-NODE.md#how-does-it-work)
```
Comment (iowillhoit) [inline DEVELOPING.md:44]: You could use the `[!CAUTION]` special blockquote syntax here if you wanted to ([docs](https://github.com/orgs/community/discussions/16925))

However, it doesn't look like this works inside of a `<detail>` block. If you want want to use it, you'll need to remove those

File: JSFORCE-NODE.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+## What is the `@jsforce/jsforce-node` package?
+
+It's a "lite" version of `jsforce` that doesn't include the browser builds and CLI deps, making it ~30M smaller that the `jsforce` package which includes all node and browser builds.
+You can use it as a drop-in replacement of `jsforce` in your Node.js project to reduce bundle size.
+
+For reference, the `jsforce` package contains:
+1. CJS & ESM builds for node
+2. ESM browser builds
+3. browser bundles + minified builds
+4. TS source code (src folder)
```
Comment (iowillhoit) [inline JSFORCE-NODE.md:12]: You could mention the CLI files here too 🤷

**Review by iowillhoit — APPROVED** [review]
A few non-blocking suggestions. Everything looks great after running commands locally 🎉

File: JSFORCE-NODE.md (hunk 2 above)
Comment (cristiand391) [reply JSFORCE-NODE.md:12]: done

**Pull request merged**